### PR TITLE
authenticate: support groups

### PR DESCRIPTION
**Pull request by desimone**
authenticate: support groups

- authenticate/providers: add group support to azure
- authenticate/providers: add group support to google
- authenticate/providers: add group support to okta
- authenticate/providers: add group support to onelogin
- {authenticate/proxy}: change default cookie lifetime timeout to 14 hours
- proxy: sign group membership
- proxy: add group header
- deployment: add CHANGELOG
- deployment: fix where make release wasn’t including version

## Related:
- #54 
- #56 

### **Checklist**:
- [x] Google provider group support
- [x] documentation updated
- [x] related issues referenced
- [x] unit tests
- [x] ready for review

### CODE REVIEW

**Comment by desimone** [comment]
I'm leaning towards adding the extra oauth2/rest calls to: 

* validate group membership on refresh
* standardize on group guid

This is a tricky one since my experiences are limited to a single identity provider. I would greatly value any feedback! 

/cc @mbarrien @victornoel @tboerger @lukasmrtvy

File: docs/docs/identity-providers.md [hunk 1]
```diff
@@ -10,15 +10,17 @@ description: >-
...
+In this guide we'll cover how to do the following for each identity provider:
+
+1. Set a **[Redirect URL]** pointing back to Pomerium.
 2. Generate a **[Client ID]** and **[Client Secret]**.
 3. Configure Pomerium to use the **[Client ID]** and **[Client Secret]** keys.
 
 ## Azure Active Directory
 
-If you plan on allowing users to log in using a Microsoft Azure Active Directory account, either from your company or from external directories, you must register your application through the Microsoft Azure portal. If you don't have a Microsoft Azure account, you can [signup](https://azure.microsoft.com/en-us/free) for free.
+If you plan on allowing users to log in using a Microsoft Azure Active Directory account, either from your company or from external directories, you must register your application through the Microsoft Azure portal. If you don't have a Microsoft Azure account, you can [signup](https://azure.c.com/en-us/free) for free.
```
Comment (victornoel) [inline docs/docs/identity-providers.md]: @desimone I think you mistakenly made a typo here :)

Comment (desimone) [reply docs/docs/identity-providers.md]: Most certainly did. Thank you!

**Comment by desimone** [comment]
* **Group Support** : The authenticate service now retrieves a user's group membership information during authentication and refresh. This change may require additional identity provider configuration; all of which are described in the [updated docs](https://www.pomerium.io/docs/identity-providers.html). A brief summary of the requirements for each IdP are as follows:
	- Google requires the [Admin SDK](https://developers.google.com/admin-sdk/directory/) to enabled, a service account with properly delegated access, and `IDP_SERVICE_ACCOUNT` to be set to the base64 encoded value of the service account's key file. 
	- Okta requires a `groups` claim to be added to both the `id_token` and `access_token`. No additional API calls are made. 
	- Microsoft Azure Active Directory requires the application be given an [additional API permission](https://docs.microsoft.com/en-us/graph/api/user-list-memberof?view=graph-rest-1.0), `Directory.Read.All`. 
	- Onelogin requires the [groups](https://developers.onelogin.com/openid-connect/scopes) was supplied during authentication and that groups parameter has been mapped. Group membership is validated on refresh with the [user-info api endpoint](https://developers.onelogin.com/openid-connect/api/user-info).

**Pull request merged**